src/Rings/MPolyMap/MPolyRing.jl

```diff
@@ -132,12 +132,47 @@ end
 #
 ################################################################################
 
+# Some additional methods needed for the test in the constructor for MPolyAnyMap
+is_gen(x::MPolyQuoRingElem) = is_gen(lift(x))
+is_gen(x::MPolyDecRingElem) = is_gen(forget_grading(x))
+
+function _evaluate_plain(F::MPolyAnyMap{<:MPolyRing, <:MPolyRing}, u)
+  if isdefined(F, :variable_indices)
+    S = codomain(F)::MPolyRing
+    r = ngens(S)
+    ctx = MPolyBuildCtx(S)
+    for (c, e) in zip(AbstractAlgebra.coefficients(u), AbstractAlgebra.exponent_vectors(u))
+      ee = [0 for _ in 1:r]
```

Comment:
```suggestion
      ee = Vector{Int}(undef, r)
```
should be an epsilon faster

Reply:
One should be able to use the "same" vector `ee` for the whole loop as it gets copied anyway. I try to optimize this further.

Reply:
An undefined vector will not work. We need the zeroes.